### PR TITLE
Update BigQueryCreateExternalTableOperator doc and parameters

### PR DESCRIPTION
**Pull request by tnyz**
Update parameter to reflect expected usage while maintaining backwards compatibility.
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Review by mik-laj — APPROVED** [review]

File: airflow/providers/google/cloud/operators/bigquery.py [hunk 1]
```diff
@@ -1142,11 +1142,14 @@ def __init__(
...
-                "Please use provide table definition using `table_resource` parameter."
-                "You can still use external `schema_object`. ",
+                "Please use provide table definition using `table_resource` parameter.",
                 DeprecationWarning,
                 stacklevel=2,
             )
+            if not bucket:
+                raise ValueError("`bucket` is required when not using `table_resource`.")
+            if not source_objects:
+                raise ValueError("`source_objects` is required when not using `table_resource`.")
```
Comment (uranusjr) [inline airflow/providers/google/cloud/operators/bigquery.py:1152]: These should be `TypeError` IMO.

Comment (turbaszek) [reply airflow/providers/google/cloud/operators/bigquery.py:1152]: I think `ValueError` is more correct.

> Passing arguments of the wrong type (e.g. passing a list when an int is expected) should result in a TypeError, but passing arguments with the wrong value (e.g. a number outside expected boundaries) should result in a ValueError.

This case is more of "wrong arguments" not "arguments types".

Comment (potiuk) [reply airflow/providers/google/cloud/operators/bigquery.py:1152]: Agree with Tomek. Value error is good here.

**Comment by potiuk** [comment]
Reopened to rebuild

**Review by turbaszek — APPROVED** [review]

**Comment by tnyz** [comment]
can someone with write access review and close this pr? thanks

**Pull request merged**